### PR TITLE
DE46989 - Fix various subtle issues with hover handling

### PR DESCRIPTION
**Pull request by devpow112**
Most of the issues were around `0` being treated as a `false` clause in `if` statements. I also fixed an issue where the hover text wouldn't display if you hovered over the "thumb" selector as well as with an issue where you added an override before selecting a level and cleared it. This would result in `0` being treated as the overridden value which was not being cleared.

### CODE REVIEW

File: src/components/rubric-criterion-mobile.js [hunk 1]
```diff
@@ -466,7 +466,10 @@ export class RubricCriterionMobile extends composeMixins(
 
 	_updateScore() {
 		const score = this.CriterionAssessmentHelper.getScore(this._assessmentCriterionEntity);
-		if (!!score && score.toString() !== this._score) {
+
+		if (score === null || score === undefined) {
```
Comment (devpow112) [inline src/components/rubric-criterion-mobile.js:470]: Fixes weird ness with override without a level selection.

File: src/components/rubric-levels-mobile.js [hunk 2]
```diff
@@ -837,8 +849,10 @@ export class RubricLevelsMobile extends composeMixins(
 		const isHovered = this._hovered >= 0 && index === this._hovered;
 		if (isHovered) {
 			const hoveredLevel = this.assessmentMap.get(this._getSelfLink(this.criterionCells[index]));
-			if (hoveredLevel && hoveredLevel.properties && hoveredLevel.properties.score) {
-				this._updateHoveredScore(hoveredLevel.properties.score);
+			const score = hoveredLevel?.properties?.score;
+
+			if (score !== null || score !== undefined) {
```
Comment (devpow112) [inline src/components/rubric-levels-mobile.js:854]: Fixed issue with `0` being treated as `false` and not displaying on hover.

File: src/components/rubric.js [hunk 3]
```diff
@@ -471,21 +472,16 @@ class Rubric extends composeMixins(
 		return this.readOnly || !this.assessmentHref;
 	}
 
-	_localizeCompactScoreText(entity, score) {
-		if (!entity?.hasOutOf) {
+	_localizeCompactScoreText() {
```
Comment (devpow112) [inline src/components/rubric.js:475]: Removed passing in of `entity` and `score`.

File: src/components/rubric.js [hunk 4]
```diff
@@ -337,7 +337,7 @@ class Rubric extends composeMixins(
 				?has-alerts="${this._hasAlerts}"
 				?compact="${this.isCompact()}"
 				?detached-view="${this.detachedView}"
-				score-text="${this._localizeCompactScoreText(this.entity, this._totalScore)}"
+				score-text="${this._localizeCompactScoreText()}"
```
Comment (devpow112) [inline src/components/rubric.js:340]: Removed passing in of `entity` and `score`.

File: src/components/rubric-editable-score.js [hunk 5]
```diff
@@ -620,14 +620,11 @@ export class RubricEditableScore extends composeMixins(
...
 		}
 
-		if (score || score === 0) {
-			return this.localize('rubricEvaluation:scoreOutOf',
-				'score', score.toString(),
-				'outOf', entity.outOf.toString()
-			);
-		} else {
-			return this.localize('rubricEvaluation:outOf', 'outOf', entity.outOf.toString());
+		if (score !== null || score !== undefined) {
```
Comment (devpow112) [inline src/components/rubric-editable-score.js:623]: Just made consistent with other `score` type `null` checks.

File: src/components/rubric-levels-mobile.js [hunk 6]
```diff
@@ -825,6 +825,18 @@ export class RubricLevelsMobile extends composeMixins(
 		this._handleDragStart(event, this._selected);
 	}
 
+	_handleThumbMouseOver() {
```
Comment (devpow112) [inline src/components/rubric-levels-mobile.js:828]: Used to fix issue with hovering over a selected level ("thumb") and it not displaying it correctly.

File: src/components/rubric.js [hunk 7]
```diff
@@ -563,8 +559,8 @@ export class Rubric extends composeMixins(
 
 		/*
 			Use .print-style-active class to set the print container and its contents as visible
-		  	Use temporary class to style the print container instead of general @media print CSS selector
-		  	This ensures only the current rubric's print container is visible (in case there are multiple rubrics in a page)
+			Use temporary class to style the print container instead of general @media print CSS selector
```
Comment (devpow112) [inline src/components/rubric.js:562]: White-space/indenting issues

**Comment by devpow112** [comment]
^ The backport stuff just seems to never work. I think it must be used with the new versioning release actions stuff and can't work without it. Will remove it until we maybe eventually update to use the LMS versioning action stuff.

**Review by imcdonald-d2l — APPROVED** [review]

**Pull request merged**